week2/project/src/AppRoutes.js

```diff
@@ -0,0 +1,66 @@
+import React from "react";
+import {
+  BrowserRouter as Router,
+  Route,
+  Routes as RoutesContainer,
+} from "react-router-dom";
+import ProductList from "./components/ProductList/ProductList";
+import ProductDetail from "./components/ProductDetail/ProductDetail";
+import CategoryList from "./components/CategoryList/CategoryList";
+
+const AppRoutes = ({
+  categories,
+  selectedCategory,
+  setSelectedProduct,
+  products,
+  handleCategoryClick,
+}) => (
+  <RoutesContainer>
+    {/* Route for displaying the details of a specific product */}
+    <Route
+      path="/product/:id"
+      element={
+        <ProductDetail
+          products={products}
+          setSelectedProduct={setSelectedProduct}
+        />
+      }
+    />
+
+    {/* Default route for displaying categories and product list */}
+    <Route
+      path="/"
+      element={
+        <React.Fragment>
+          {/* Component for displaying a list of categories */}
+          <CategoryList
+            categories={categories}
+            selectedCategory={selectedCategory}
+            onCategoryClick={handleCategoryClick}
+          />
+
+          {/* Component for displaying a list of products */}
```

Comment:
The comments are not necessary because the components already explain what happens here. Which means you write clear code and as another developer and reader of your code I clearly understand what is going on 👍 

```suggestion
          <CategoryList
            categories={categories}
            selectedCategory={selectedCategory}
            onCategoryClick={handleCategoryClick}
          />
```